mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,225 @@
+                    service.stage_block(block_id, chunk, headers=headers)
+                uploading_block_list.append(block_id)
+            try:
+                service.commit_block_list(uploading_block_list)
+            except ClientAuthenticationError:
+                _logger.warning(
+                    "Failed to authorize request, possibly due to credential expiration."
+                    "Refreshing credentials and trying again..")
+                credentials = self._get_write_credentials(self.run_id,
+                                                          artifact_path).credentials.signed_uri
+                service = BlobClient.from_blob_url(blob_url=credentials, credential=None)
+                service.commit_block_list(uploading_block_list)
+        except Exception as err:
+            raise MlflowException(err)
+
+    def _aws_upload_file(self, credentials, local_file):
+        try:
+            headers = self._extract_headers_from_credentials(credentials)
+            signed_write_uri = credentials.signed_uri
+            with open(local_file, 'rb') as file:
+                put_request = requests.put(signed_write_uri, headers=headers, data=file)
+                put_request.raise_for_status()
+        except Exception as err:
+            raise MlflowException(err)
+
+    def _upload_to_cloud(self, cloud_credentials, local_file, artifact_path):
+        if cloud_credentials.credentials.type == ArtifactCredentialType.AZURE_SAS_URI:
+            self._azure_upload_file(cloud_credentials.credentials, local_file, artifact_path)
+        elif cloud_credentials.credentials.type == ArtifactCredentialType.AWS_PRESIGNED_URL:
+            self._aws_upload_file(cloud_credentials.credentials, local_file)
+        else:
+            raise MlflowException('Not implemented yet')
+
+    def _download_from_cloud(self, cloud_credential, local_file_path):
+        """
+        Downloads a file from the input `cloud_credential` and save it to `local_path`.
+
+        Since the download mechanism for both cloud services, i.e., Azure and AWS is the same,
+        a single download method is sufficient.
+
+        The default working of `requests.get` is to download the entire response body immediately.
+        However, this could be inefficient for large files. Hence the parameter `stream` is set to
+        true. This only downloads the response headers at first and keeps the connection open,
+        allowing content retrieval to be made via `iter_content`.
+        In addition, since the connection is kept open, refreshing credentials is not required.
+        """
+        if cloud_credential.type not in [ArtifactCredentialType.AZURE_SAS_URI,
+                                         ArtifactCredentialType.AWS_PRESIGNED_URL]:
+            raise MlflowException(message='Cloud provider not supported.',
+                                  error_code=INVALID_PARAMETER_VALUE)
```

Comment:
Nice exception! I think the error code here should be `INTERNAL_ERROR` in this case (the default error code), since the user has no control over this failure mode. If the MLflow service suddenly starts passing strange credential types to the client, there's nothing the end user can do about it.

Reply:
Done.